bin/client/src/interop/transition.rs

```diff
@@ -98,14 +102,31 @@ where
             .await?;
     l2_provider.set_cursor(cursor.clone());
 
-    let pipeline = OraclePipeline::new(
+    let mut pipeline = OraclePipeline::new(
         rollup_config.clone(),
         cursor.clone(),
         oracle.clone(),
         beacon,
         l1_provider.clone(),
         l2_provider.clone(),
     );
+
+    // Reset the pipeline to populate the initial system configuration in L1 Traversal.
+    let l2_safe_head = *cursor.read().l2_safe_head();
+    pipeline
+        .signal(
+            ResetSignal {
+                l2_safe_head,
+                l1_origin: cursor.read().origin(),
+                system_config: l2_provider
+                    .system_config_by_number(l2_safe_head.block_info.number, rollup_config.clone())
+                    .await
+                    .ok(),
+            }
+            .signal(),
+        )
+        .await?;
```

Comment:
This looks to be duplicated. I think I'd prefer to introduce a new method on the `Pipeline` trait to populate the initial system config. Otherwise, it is easy user error to _forget_ to do this.

Reply:
Yeah can do.

Reply:
Updated to hide the reset in the constructor. Good?